packages/backend-lib/src/types.ts

```diff
@@ -146,6 +146,238 @@ export const DecodedJwt = Type.Object({
+  callerIdentity: Type.String(),
+  destination: Type.Array(Type.String()),
+  headers: Type.Optional(
+    Type.Array(
+      Type.Object({
+        name: Type.String(),
+        value: Type.String(),
+      }),
+    ),
+  ),
+  headersTruncated: Type.Optional(Type.Boolean()),
+  commonHeaders: Type.Optional(
+    Type.Object({
+      from: Type.Array(Type.String()),
+      to: Type.Array(Type.String()),
+      date: Type.String(),
+      messageId: Type.String(),
+      subject: Type.String(),
+    }),
+  ),
+  tags: Type.Record(Type.String(), Type.Array(Type.String())),
+});
+
+export type AmazonSesMailData = Static<typeof AmazonSesMailData>;
+
+export const AmazonSesClickEvent = Type.Object({
+  eventType: Type.Literal(AmazonSesNotificationType.Click),
+  mail: AmazonSesMailData,
+  click: Type.Object({
+    ipAddress: Type.String(),
+    timestamp: Type.String(),
+    userAgent: Type.String(),
+    link: Type.String(),
+    linkTags: Type.String(), // This type may not be correct
+  }),
+});
+
+export const AmazonSesOpenEvent = Type.Object({
+  eventType: Type.Literal(AmazonSesNotificationType.Open),
+  mail: AmazonSesMailData,
+  open: Type.Object({
+    ipAddress: Type.String(),
+    timestamp: Type.String(),
+    userAgent: Type.String(),
+  }),
+});
+
+export const AmazonSesSendEvent = Type.Object({
+  eventType: Type.Literal(AmazonSesNotificationType.Send),
+  mail: AmazonSesMailData,
+});
+
+export const AmazonSesRejectEvent = Type.Object({
+  eventType: Type.Literal(AmazonSesNotificationType.Reject),
+  mail: AmazonSesMailData,
+  reject: Type.Object({
+    reason: Type.String(),
+  }),
+});
+
+export const AmazonSesBounceEvent = Type.Object({
+  eventType: Type.Literal(AmazonSesNotificationType.Bounce),
+  mail: AmazonSesMailData,
+  bounce: Type.Object({
+    bounceType: Type.Enum(AmazonSesBounceType),
+    bounceSubType: Type.Enum(AmazonSesBounceSubType),
+    bouncedRecipients: Type.Array(
+      Type.Object({
+        emailAddress: Type.String(),
+        action: Type.Optional(Type.String()),
+        status: Type.Optional(Type.String()),
+        diagnosticCode: Type.Optional(Type.String()),
+      }),
+    ),
+    timestamp: Type.String(),
+    feedbackId: Type.String(),
+    remoteMtaIp: Type.Optional(Type.String()),
+    reportingMTA: Type.Optional(Type.String()),
+  }),
+});
+
+export const AmazonSesComplaintEvent = Type.Object({
+  eventType: Type.Literal(AmazonSesNotificationType.Complaint),
+  mail: AmazonSesMailData,
+  complaint: Type.Object({
+    complainedRecipients: Type.Array(
+      Type.Object({
+        email: Type.String(),
+      }),
+    ),
+    timestamp: Type.String(),
+    feedbackId: Type.String(),
+    complaintSubType: Type.Enum(AmazonSesComplaintSubType),
+    userAgent: Type.Optional(Type.String()),
+    complaintFeedbackType: Type.Optional(Type.String()),
+    arrivalDate: Type.Optional(Type.String()),
+  }),
+});
+
+export const AmazonSesDeliveryEvent = Type.Object({
```

Comment:
A lot of thorough typing here thank you!

Reply:
I was hoping it wasn't too thorough :-)